core/builder/src/main/java/io/quarkus/builder/Execution.java

```diff
@@ -56,12 +68,36 @@
             done = true;
     }
 
+    Map<StepInfo, Integer> orderTopologically(Set<StepInfo> startSteps) {
+        if (!(startSteps instanceof SortedSet)) {
+            throw new IllegalArgumentException("startSteps must be a SortedSet");
+        }
+
+        /* Kahn's algorithm */
+        final Map<StepInfo, AtomicInteger> indegrees = new HashMap<StepInfo, AtomicInteger>();
```

Comment:
This is another place where it seems like a field on `StepInfo` would be simpler and faster.

Reply:
+1